### PR TITLE
Form & Input validation

### PR DESCRIPTION
**Pull request by djarufes**
Validation added to the following:
- Login email/password
- Register email/password
- Profile display name change
- Profile Change password
- Create Group name/description
- Chat component input box

### CODE REVIEW

**Comment by harmonr** [comment]
Profile:
The red text when changing the profile name must have more than one character, needs a space between "At least" not "Atleast"

**Comment by harmonr** [comment]
Profile:
When changing password, original password (the actual original password) is in the form control so you can view it if you click the icon.

File: src/app/profile/profile.component.html [hunk 1]
```diff
@@ -68,36 +80,102 @@
...
             mat-stroked-button
             type="submit"
+            (click)="onPasswordChange()"
             [disabled]="!passwordForm.valid"
             style="margin-left: 115px"
           >
             Save
           </button>
           <button mat-stroked-button (click)="cancelPassword()">Cancel</button>
+          <div class="container">
```
Comment (mburakyucel) [inline src/app/profile/profile.component.html:161]: Do we need a container here just to display messages

File: src/app/profile/profile.component.ts [hunk 2]
```diff
@@ -19,12 +26,28 @@ export class ProfileComponent implements OnInit, OnDestroy {
...
-    old_password: new FormControl(''),
-    new_password: new FormControl(''),
-    confirm_password: new FormControl(''),
-  });
+  public displayName: any = new FormControl('', [
+    Validators.required,
+    Validators.maxLength(32),
+  ]);
+  public tempDisplayName: any;
+  public passwordForm: any = this.formBuilder.group(
```
Comment (mburakyucel) [inline src/app/profile/profile.component.ts:34]: Why did we remove FormGroup and replaced with formBuilder? I also see a deprecation warning for formbuilder.group

File: src/app/profile/profile.component.ts [hunk 3]
```diff
@@ -19,12 +26,28 @@ export class ProfileComponent implements OnInit, OnDestroy {
...
-  public passwordForm: any = new FormGroup({
-    old_password: new FormControl(''),
-    new_password: new FormControl(''),
-    confirm_password: new FormControl(''),
-  });
+  public displayName: any = new FormControl('', [
+    Validators.required,
+    Validators.maxLength(32),
+  ]);
+  public tempDisplayName: any;
```
Comment (mburakyucel) [inline src/app/profile/profile.component.ts:33]: What is this tempDisplayName?

**Review by mburakyucel — APPROVED** [review]

**Pull request merged**